demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -115,6 +149,83 @@
  */
 #define DELAY_BETWEEN_DEMO_RETRY_ITERATIONS_S    ( 5 )
 
+/**
+ * @brief Length in bytes of hash digest.
+ */
+#define SHA256_HASH_DIGEST_LENGTH                ( ( ( uint16_t ) 64 ) )
+
+/**
+ * @brief Buffer Length for AWS IOT Credentials.
+ */
+#define CREDENTIAL_BUFFER_LENGTH                 2048U
+
+/**
+ * @brief AWS Service name to send HTTP request using SigV4 library.
+ */
+#define AWS_SERVICE_NAME                         "s3"
+
+/**
+ * @brief Full AWS S3 server endpoint.
+ */
+#define AWS_S3_OBJECT_URL                       \
+    "https://"                                  \
+    AWS_S3_BUCKET_NAME "." AWS_SERVICE_NAME "." \
+    AWS_S3_BUCKET_REGION  ".amazonaws.com/"     \
+    AWS_S3_OBJECT_NAME
+
+/**
+ * @brief Full credentials endpoint including role alias.
+ */
+#define AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT         \
+    "https://"                                            \
+    AWS_IOT_CREDENTIAL_PROVIDER_ENDPOINT "/role-aliases/" \
+    AWS_IOT_CREDENTIAL_PROVIDER_ROLE "/credentials"
+
+/**
+ * @brief Field name of the HTTP thing name header sent in HTTP request to AWS S3.
+ */
+#define AWS_IOT_THING_NAME_HEADER_FIELD         "x-amz-iot-thing-name"
+
+/**
+ * @brief Field name of the HTTP date header to read from the AWS IOT credential provider server response.
+ */
+#define HTTP_DEMO_RECEIVED_DATE_HEADER_FIELD    "date"
+
+/**
+ * @brief Field name of the HTTP Authorization header to add to the request headers.
+ */
+#define SIGV4_AUTH_HEADER_FIELD_NAME            "Authorization"
+
+/**
+ * @brief IS8601 formatted date length.
+ */
+#define SIGV4_ISO_STRING_LEN                    16U
+
+/**
+ * @brief Length of AWS HTTP Authorization header value generated using SigV4 library.
+ */
+#define AWS_HTTP_AUTH_HEADER_VALUE_LEN          2048U
+
+/**
+ * @brief Length for AWS IOT Credential provider server host name.
+ */
+#define SERVER_HOST_NAME_LENGTH                 65U
+
+/**
+ * @brief Hex encoded hash digest length.
+ */
+#define SHA256_HEX_ENCODED_DIGEST_LENGTH        ( ( ( uint16_t ) 64 ) )
+
+/**
+ * @brief Maximum block size of SHA256 Hashing Algorithm used in the demo.
+ */
+#define SHA256_HASH_MAX_BLOCK_LENGTH            64U
+
+/**
+ * @brief Maximum digest length of the SHA256 hash function used in the demo.
+ */
+#define SHA256_HASH_MAX_DIGEST_LENGTH           ( SHA256_HEX_ENCODED_DIGEST_LENGTH / 2 )
```

Comment:
This seems redundant to `SHA256_HASH_DIGEST_LENGTH` macro (once corrected to 32) defined earlier